### PR TITLE
Hashed repo path

### LINKED ISSUE #38: Store repositories as hashes
TL;DR; instead of storing the repos under `storage/{username}/{reponame}` we should have a on-create-time immutable (not possible in cockroach I know...) name that we use instead.

For example `hashAlgo({username}/{reponame}-{creation_time})` should suffice. Then on disk we'd go e.g. `storage/{hash[:2]}/{hash[2:2]}/{hash[4:]}` (if my array-math checks out...)

Why?
- No need to rename on platter when renaming a repo, or moving to another "group"/namespace. Just a DB-call and we're done.

### PR DESCRIPTION
**Pull request by bkcsoft**
Uses the table-rows ID (UUID) as a hash... not sure if I like that or not...

Skipped the `old_repo_name`-table as we don't have anything that renames Repos yet...

Closes #38

### CODE REVIEW

**Comment by metalmatze** [comment]
This is still missing a database migration with a new table and some more columns to the repositories table, is it?
Also it needs are rebase :relaxed:

**Comment by bkcsoft** [comment]
@metalmatze No need for a new table or migration, we can add that when we add support for renaming repos :slightly_smiling_face:

File: cmd/storage/cli.go [hunk 1]
```diff
@@ -14,17 +14,17 @@ import (
 
 func treeAction(c *cli.Context) error {
 	args := c.Args()
-	if len(args) != 4 {
-		return errors.New("need exactly 4 arguments: owner name rev path")
+	if len(args) != 3 {
+		return errors.New("need exactly 3 arguments: repo_hash rev path")
```
Comment (metalmatze) [inline cmd/storage/cli.go]:
```suggestion
		return errors.New("need exactly 3 arguments: id ref path")
```

File: cmd/storage/cli.go [hunk 2]
```diff
@@ -14,17 +14,17 @@ import (
...
 	}
-	owner, name, rev, path := args[0], args[1], args[2], args[3]
+	repoHash, rev, path := args[0], args[1], args[2]
```
Comment (metalmatze) [inline cmd/storage/cli.go]:
```suggestion
	id, rev, path := args[0], args[1], args[2]
```

File: pkg/sourcepods/repository/service.go [hunk 3]
```diff
@@ -22,27 +22,27 @@ type (
...
 	}
 
 	// Storage manages the git storage
 	Storage interface {
-		Create(ctx context.Context, owner string, name string) error
-		SetDescription(ctx context.Context, owner, name, description string) error
-		Branches(ctx context.Context, owner string, name string) ([]storage.Branch, error)
-		Commit(ctx context.Context, owner string, name string, rev string) (storage.Commit, error)
-		Tree(ctx context.Context, owner string, name string, rev string, path string) ([]storage.TreeEntry, error)
+		Create(ctx context.Context, repoPath string) error
```
Comment (metalmatze) [inline pkg/sourcepods/repository/service.go]:
```suggestion
		Create(ctx context.Context, id string) error
```

Comment (metalmatze) [reply pkg/sourcepods/repository/service.go]: Some for the others below

File: pkg/storage/client.go [hunk 4]
```diff
@@ -35,46 +35,36 @@ func NewClient(storageAddr string) (*Client, error) {
 }
 
 // Create a repository
-func (c *Client) Create(ctx context.Context, owner, name string) error {
+func (c *Client) Create(ctx context.Context, repoHash string) error {
```
Comment (metalmatze) [inline pkg/storage/client.go]:
```suggestion
func (c *Client) Create(ctx context.Context, id string) error {
```

File: pkg/storage/client.go [hunk 5]
```diff
@@ -35,46 +35,36 @@ func NewClient(storageAddr string) (*Client, error) {
...
 	span, ctx := opentracing.StartSpanFromContext(ctx, "storage.Client.Create")
-	span.SetTag("owner", owner)
-	span.SetTag("name", name)
+	span.SetTag("repo_hash", repoHash)
```
Comment (metalmatze) [inline pkg/storage/client.go]:
```suggestion
	span.SetTag("id", id)
```

File: pkg/storage/client.go [hunk 6]
```diff
@@ -35,46 +35,36 @@ func NewClient(storageAddr string) (*Client, error) {
...
-		Owner: owner,
-		Name:  name,
-	})
+	_, err := c.repos.Create(ctx, &CreateRequest{RepoHash: repoHash})
 	return err
 }
 
 // SetDescription of a repository
-func (c *Client) SetDescription(ctx context.Context, owner, name, description string) error {
+func (c *Client) SetDescription(ctx context.Context, repoHash, description string) error {
```
Comment (metalmatze) [inline pkg/storage/client.go]:
```suggestion
func (c *Client) SetDescription(ctx context.Context, id, description string) error {
```

File: pkg/storage/client.go [hunk 7]
```diff
@@ -35,46 +35,36 @@ func NewClient(storageAddr string) (*Client, error) {
...
-		Name:        name,
+		RepoHash:    repoHash,
 		Description: description,
 	})
 	return err
 }
 
 // Branches returns all branches of a repository
-func (c *Client) Branches(ctx context.Context, owner, name string) ([]Branch, error) {
+func (c *Client) Branches(ctx context.Context, repoHash string) ([]Branch, error) {
```
Comment (metalmatze) [inline pkg/storage/client.go]:
```suggestion
func (c *Client) Branches(ctx context.Context, id string) ([]Branch, error) {
```

File: pkg/storage/client.go [hunk 8]
```diff
@@ -89,17 +79,15 @@ func (c *Client) Branches(ctx context.Context, owner, name string) ([]Branch, er
 }
 
 // Commit returns a single commit from a given repository
-func (c *Client) Commit(ctx context.Context, owner string, name string, ref string) (Commit, error) {
+func (c *Client) Commit(ctx context.Context, repoHash, ref string) (Commit, error) {
```
Comment (metalmatze) [inline pkg/storage/client.go]:
```suggestion
func (c *Client) Commit(ctx context.Context, id, ref string) (Commit, error) {
```

File: pkg/storage/storage.proto [hunk 9]
```diff
@@ -19,19 +19,16 @@ service Commit {
 }
 
 message CreateRequest {
-    string owner = 1;
-    string name = 2;
+    string repo_hash = 1;
```
Comment (metalmatze) [inline pkg/storage/storage.proto]:
```suggestion
    string id = 1;
```

File: pkg/storage/storage.proto [hunk 10]
```diff
@@ -19,19 +19,16 @@ service Commit {
...
 }
 
 message SetDescriptionRequest {
-    string owner = 1;
-    string name = 2;
-    string description = 3;
+    string repo_hash = 1;
```
Comment (metalmatze) [inline pkg/storage/storage.proto]:
```suggestion
    string id = 1;
```

File: pkg/storage/storage.proto [hunk 11]
```diff
@@ -19,19 +19,16 @@ service Commit {
...
+    string description = 2;
 }
 
 message BranchesRequest {
-    string owner = 1;
-    string name = 2;
+    string repo_hash = 1;
```
Comment (metalmatze) [inline pkg/storage/storage.proto]:
```suggestion
    string id = 1;
```

Comment (metalmatze) [reply pkg/storage/storage.proto]: same below

**Pull request merged**